src/App.tsx

```diff
@@ -1,14 +1,52 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
+import { getTodos, getUser } from './api';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import { getFilteredTodos } from './utils/getFilteredTodos';
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filteredStatus, setFilteredStatus] = useState('all');
+  const [searchQuery, setSearchQuery] = useState('');
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    const loadTodos = () => {
+      getTodos().then(res => {
+        setTodos(res);
+        setIsLoading(false);
+      });
+    };
+
+    loadTodos();
+  }, []);
+
+  const handleShowTodo = (todo: Todo) => {
+    setIsLoading(true);
+    setSelectedTodo(todo);
+    getUser(todo.userId).then(res => setUser(res));
+
+    setIsLoading(false);
+  };
+
+  const handleHideTodo = () => {
+    setSelectedTodo(null);
+    setUser(null);
+  };
```

Comment:
You can use single function - something like this:


```suggestion
  const toggleTodo = (todo?: Todo) => {
  if (todo)
    {setIsLoading(true);
    setSelectedTodo(todo);
    getUser(todo.userId).then(res => setUser(res));

    setIsLoading(false);
    } else  {
    setSelectedTodo(null);
    setUser(null);
  };
  }
```

Reply:
JFYI `?` in arguments stands for the optional argument